### DISCUSSION TITLE
Sourcemap URL incorrect when bundling webpack

### DISCUSSION THREAD

**Discussion by a-gehlot** [webpack/webpack#16580] [Q&A]
# Bug report

**What is the current behavior?**
When attempting to find source maps, console shows

> "DevTools failed to load source map: Could not load content for http://localhost:3000/assets/bundle.js-7f6807a0f15667890c1616b3417f8d5e6706c2c168dadbd124f0c1de47d079ba.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE"

**If the current behavior is a bug, please provide the steps to reproduce.**
In webpack.config.js, include     
`devtool: "source-map"`
Run `webpack --watch --mode=development`.

Upon inspecting bundle.js, the issue is that the `sourceMappingURL` is not pointing to the directory. It shows 
`//# sourceMappingURL=bundle.js.map`, while it should be showing
`//# sourceMappingURL=./bundle.js.map`

Manually changing the URL makes the source maps work, but if I rebundle, it will revert back to the error.

**What is the expected behavior?**
The sourceMappingURL is correct and the source maps load within the Chrome console.

**Other relevant information:**
webpack version:  5.75.0
Node.js version: 16.15.0
Operating System: MacOS Ventura 13.1 (note: this could potentially be caused by this recent update)
Additional tools:

**Comment by alexander-akait** [comment]
Please provide your configuration, I think you have wrong public path

**Reply by a-gehlot** [reply]
My webpack config file is the following:

```
const path = require('path');

module.exports = {
    context: __dirname,
    entry: './frontend/pokedex.jsx',
    output: {
        path: path.resolve(__dirname, 'app', 'assets', 'javascripts'),
        filename: 'bundle.js'
    },
    resolve: {
        extensions: ['.js', '.jsx', '*']
    },
    module: {
        rules: [
            {
                test: /\.jsx?$/,
                exclude: /(node_modules)/,
                use: {
                    loader: 'babel-loader',
                    options: {
                        presets: ['@babel/env', '@babel/react']
                    }
                },
            }
        ]
    },
    devtool: "source-map"
};
```

**Reply by alexander-akait** [reply]
Do you have somthing special for babel?

`//# sourceMappingURL=bundle.js.map` and `//# sourceMappingURL=./bundle.js.map` are the same if you don't use bundle for multiple pages (and pages are nested)

**Reply by a-gehlot** [reply]
I don't believe so, but here are my dependencies if that helps anything:

```
  "dependencies": {
    "@babel/core": "^7.20.2",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@reduxjs/toolkit": "^1.9.0",
    "babel-loader": "^9.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.0.5",
    "react-router-dom": "^6.4.3",
    "redux": "^4.2.0",
    "redux-logger": "^3.0.6",
    "webpack": "^5.75.0",
    "webpack-cli": "^5.0.0"
  }
```

I did just try again to delete the `bundle.js` file and re-run webpack, but it again couldn't find the source-maps in the newly generated file, until I added `./`, and they are nested in the same folder. If any of this helped, here's what the webpack showed on console:

```
asset bundle.js 1.56 MiB [compared for emit] (name: main) 1 related asset
runtime modules 1.25 KiB 6 modules
modules by path ./node_modules/ 1.47 MiB 56 modules
modules by path ./frontend/ 7.97 KiB
  modules by path ./frontend/reducers/*.js 2.34 KiB
    ./frontend/reducers/selectors.js 115 bytes [built] [code generated]
    + 6 modules
  modules by path ./frontend/components/ 3.17 KiB
    modules by path ./frontend/components/pokemon/*.jsx 2.19 KiB 4 modules
    + 2 modules
  modules by path ./frontend/*.jsx 1.19 KiB
    ./frontend/pokedex.jsx 810 bytes [built] [code generated]
    ./frontend/root.jsx 406 bytes [built] [code generated]
  ./frontend/util/api_util.js 259 bytes [built] [code generated]
  ./frontend/actions/pokemon_actions.js 875 bytes [built] [code generated]
  ./frontend/store/store.js 180 bytes [built] [code generated]
webpack 5.75.0 compiled successfully in 2550 ms
```

**Reply by alexander-akait** [reply]
Can you provide `public`/`dist` structure?

**Reply by a-gehlot** [reply]
Does this work? https://github.com/a-gehlot/Pokedex

**Reply by alexander-akait** [reply]
Sorry for big delay, as you can see I was not active, I don't see any problems with source map generation, can you provide screenshot of the problem, there are two potential problems:
1. Webserver configuration
2. You load js something unusual, like from nested page

I don't have and don't know ruby, so it is hard to me run your application, if you will provide instructions how to do it, I will look deeply
